modules/bam2fastq/1.2/bam2fastq.smk

```diff
@@ -15,6 +15,25 @@
 # Import package with useful functions for developing analysis modules
 import oncopipe as op
 
+# Check that the oncopipe dependency is up-to-date. Add all the following lines to any module that uses new features in oncopipe
+min_oncopipe_version="1.0.11"
+import pkg_resources
+try:
+    from packaging import version
+except ModuleNotFoundError:
+    sys.exit("The packaging module dependency is missing. Please install it ('pip install packaging') and ensure you are using the most up-to-date oncopipe version")
+
+# To avoid this we need to add the "packaging" module as a dependency for LCR-modules or oncopipe
+
+current_version = pkg_resources.get_distribution("oncopipe").version
+if version.parse(current_version) < version.parse(min_oncopipe_version):
+    print('\x1b[0;31;40m' + f'ERROR: oncopipe version installed: {current_version}' + '\x1b[0m')
+    print('\x1b[0;31;40m' + f"ERROR: This module requires oncopipe version >= {min_oncopipe_version}. Please update oncopipe in your environment" + '\x1b[0m')
```

Comment:
Please use logs rather than the print statement. This one should output a warning if the version is wrong.

Reply:
would this work
import logging

logging.debug('This is a debug message')
logging.info('This is an info message')
logging.warning('This is a warning message')
logging.error('This is an error message')

Reply:
Yes but it should be a warning not an error. You can use the warnings.warn() function.
logging.error is to report suppression of an error without raising an exception. If you think the program should be halted if the version is not up to date then you should raise an exception instead of a warning.

Reply:
Updated version :  

```python
import warnings

warnings.simplefilter('error', UserWarning)

print('Before the warning')
warnings.warn('\x1b[0;31;40m' + f'ERROR: oncopipe version installed: yggi' +"\n\n" f"ERROR: This module requires oncopipe version >= o0jpo. Please update oncopipe in your environment" + '\x1b[0m')
```

Output:
```python
Before the warning
---------------------------------------------------------------------------
UserWarning                               Traceback (most recent call last)
<ipython-input-9-013c74db9f84> in <module>
      4 
      5 print('Before the warning')
----> 6 warnings.warn('\x1b[0;31;40m' + f'ERROR: oncopipe version installed: yggi' +"\n\n" f"ERROR: This module requires oncopipe version >= o0jpo. Please update oncopipe in your environment" + '\x1b[0m')

UserWarning: ERROR: oncopipe version installed: yggi

ERROR: This module requires oncopipe version >= o0jpo. Please update oncopipe in your environment
```

Reply:
You should use the same logger as Snakemake with `from snakemake.logging import logger`. Here's an [example use](https://github.com/LCR-BCCRC/lcr-modules/blob/master/oncopipe/oncopipe/__init__.py#L60-L64) of `logging.warning()`.

Reply:
Done

Reply:
Thanks Bruno.